pkg/kubeapiserver/authenticator/config.go

```diff
@@ -90,7 +97,7 @@ type Config struct {
 
 // New returns an authenticator.Request or an error that supports the standard
 // Kubernetes authentication mechanisms.
-func (config Config) New() (authenticator.Request, *spec.SecurityDefinitions, spec3.SecuritySchemes, error) {
+func (config Config) New() (authenticator.Request, func(*apiserver.AuthenticationConfiguration) error, *spec.SecurityDefinitions, spec3.SecuritySchemes, error) {
```

Comment:
trying to wrap my head around this structure... 
* this returns an authenticator based on the current content of `Config`
* this allows for the `AuthenticationConfiguration` (and only the `AuthenticationConfiguration`) to be updated
* this makes updating the `AuthenticationConfiguration` (and bumping of metrics related to reload?) the responsibility of the caller

we want to transition more things into AuthenticationConfiguration in the future, right? how would that work when the same things can be represented in fields in config (which this structure doesn't make dynamic) OR in the authentication configuration?

Reply:
> In the same way that we only represent the OIDC CLI flags in `AuthenticationConfiguration` now, as new things become configurable via that file in the future, we would move them into the `AuthenticationConfiguration` struct and remove them from `Config`. It would not be as clean as the OIDC flag since we were able to make those mutually exclusive, i.e. we would have to track which things started out being configured via flags vs. the file and only reload the file if it did not try to override the flag based config with the reloaded file based config.

Ok, I can envision what that would look like... we'd probably augment the config validation with some mutually-exclusive-with flag checks (can't put token webhooks in the file _and_ in the flags, etc), run that same validation on file reload, and add in the flag-specified bits to the loaded-from-file config.

> I can pass a context into New and have it handle the reload logic as well, that may end up being easier to reason about long term.

If we'll ever have to merge flag and file config (which I think we will since we started the file config unable to express all authenticators), it probably makes sense to keep the reload logic outside config.New nearer to the thing that knows about the flags